src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -1428,8 +1503,31 @@ private int InnerProcessTraceFileMain()
                             continue;
                         }
 
-                        if ((e.TimeStampRelativeMSec >= excludeEventsBefore) && (e.TimeStampRelativeMSec <= excludeEventsAfter))
-                            methodsToAttemptToPrepare.Add((int)e.EventIndex, new ProcessedMethodData(e.TimeStampRelativeMSec, method, "JitStart"));
+                        if (e.TimeStampRelativeMSec < excludeEventsBefore)
+                        {
+                            continue;
+                        }
+
+                        if (e.TimeStampRelativeMSec > excludeEventsAfter)
+                        {
+                            break;
+                        }
+
+                        if (includeMethods != null || excludeMethods != null)
+                        {
+                            string methodAsString = method.ToString();
+                            if (includeMethods != null && !includeMethods.IsMatch(methodAsString))
+                            {
+                                continue;
+                            }
+
+                            if (excludeMethods != null && excludeMethods.IsMatch(methodAsString))
+                            {
+                                continue;
+                            }
+                        }
+
+                        methodsToAttemptToPrepare.Add((int)e.EventIndex, new ProcessedMethodData(e.TimeStampRelativeMSec, method, "JitStart"));
```

Comment:
Introduce a small helper for this? (I realize it was already somewhat duplicative)

Reply:
Should the helper be for the entire for loop or just this section?
The duplicative logic is the logic within `if (_command.ProcessR2REvents)` and the logic within `if (_command.ProcessJitEvents)` right?

For either case, the two differ by the underlying type of `e`, and I'm not sure what a good wrapper type to represent both
```
[Microsoft.Diagnostics.Tracing.TraceEvent]Microsoft.Diagnostics.Tracing.Parsers.Clr.R2RGetEntryPointTraceData
```
and
```
[Microsoft.Diagnostics.Tracing.TraceEvent]Microsoft.Diagnostics.Tracing.Parsers.Clr.MethodJittingStartedTraceData
```

Reply:
Added `PassesMethodFilter`